---
layout: default
tags:
title: Differentially Expressed Genes in Schizophrenia
featimg: "venn-triple-1.png"
---

This analysis compares some gene sets raised by our study to the set of differential expressed genes in schizophrenia (SCZ) presented by ([Fromer et al 2016](http://www.nature.com/neuro/journal/vaop/ncurrent/full/nn.4399.html)) in their article *Gene expression elucidates functional impact of polygenic risk for schizophrenia*.  For discussion see the post [about affected genes]({{ site.baseurl }}{% post_url /projects/monoallelic-brain/2016-10-19-about-affected-genes %}).

```{r echo=FALSE, warnings=FALSE}
library(VennDiagram)
opts_chunk$set(dpi = 144)
opts_chunk$set(out.width = "700px")
opts_chunk$set(dev = c("png", "pdf"))
```

## Import data

```{r}
diff.e.genes <- read.csv("../../data/fromer-2016-nat-neurosci/nn.4399-S5.csv", skip = 1)
our.genes <- list()
# the selected genes that were analyzed using regression
our.genes$selected <- as.character(read.csv("../../data/genes.regression.new")[[1]])
# the genes with significant association to some biological predictor
our.genes$signif <- as.character(read.csv("../../results/signif-gene-effects-either.csv")[[1]])
```

## Results

The following genes were found to be differentially expressed by Fromer et al **AND**

1. either were merely selected for analysis in our study
1. or, furthermore, we found their parental expression bias to be significantly associated to some biological predictor

```{r}
(both <- lapply(our.genes, intersect, levels(diff.e.genes$Gene.Symbol)))
```

The characteristics of the expression change for the selected genes:

```{r}
diff.e.genes[diff.e.genes$Gene.Symbol %in% both$selected, c("Gene.Symbol", "logFC", "p.value")]
```

Further information on the significant gene(s):

```{r}
read.csv("../../results/signif-gene-effects-either-manual-annot.csv", row.names = 1)[both$signif, ]
```

```{r venn-triple}
grid.draw(venn.diagram(list(diff.e.genes$Gene.Symbol, our.genes$selected, c("RP11-909M7.3", "PEG10", "MEST", "UBE3A")), filename=NULL, category = c("SCZ: overall expression", "called imprinted", "SCZ: parental bias"), ext.text = FALSE, cat.pos = c(-15, 15, 15), cat.cex = rep(1.2, 3), col = my.col <- c("darkgreen", "blue", "red"), fill = my.col, cat.col = my.col))
```

Out of the `r length(our.genes$signif)` genes that our study found to be significantly associated to some biological predictor in terms of allelic bias, only `r length(both$signif)` of them, `r both$signif`, was/were found to be differentially expressed by Fromer et al.
