## Motivation

Imprinted genes form **clusters** of one or more genes.  Previous work by Ifat established the **imprinting status** of each gene as either known to be imprinted, *not* known to be imprinted but near an "known" gene, or else neither.  I refer to these three categories as "known", "nearby candidate" and "candidate", respectively.

The main question queries the mechanism of the age effect on imprinting (loss, gain or lack of effect).  In particular: does age regulate genes within some cluster in a concerted or an independent manner?

The current work studies this by performing two main steps

1. delineation of clusters
1. visualize clusters and age effect in terms of regression coefficients for age

## Data import and preparation

```{r echo=FALSE}
library(lattice)
library(latticeExtra)
opts_chunk$set(dpi = 144)
opts_chunk$set(out.height = "700px")
opts_chunk$set(dev = c("png", "pdf"))
lattice.options(default.args = list(as.table = TRUE))
lattice.options(default.theme = "standard.theme")
lattice.options(default.args = list(as.table = TRUE))
```

Load data importer functions:
```{r}
source("../../src/import-data.R")
source("../../src/utils.R")
source("~/projects/monoallelic-brain/src/fit-glms.R")
source("2016-08-08-imprinted-gene-clusters.R")
```

### Genome-wide data

```{r cache=TRUE}
gene.summary <-
    read.csv("../../data/readcount/summary-all-genes.csv")
rownames(gene.summary) <- gene.summary$Symbol
gene.summary$file.size <-
    read.csv("../../data/readcount/fsize-genes.csv", row.names = 2)[rownames(gene.summary), , drop = TRUE]
gene.ids <- with(gene.summary, as.character(Symbol)[ file.size > 0 ])
genes2fit <- unlist(read.csv("../../data/genes.regression.new", as.is = TRUE))
names(genes2fit) <- genes2fit
```

```{r genomwide-data-import, cache=TRUE, warning=FALSE}
Y <- get.readcounts(gene.ids, g.subsets = list(), sel.var = c("H", "N"))
S <- data.frame(lapply(gene.ids, function(g) Y[[g]]$H / Y[[g]]$N), check.names = FALSE)
names(S) <- gene.ids
N <- data.frame(lapply(Y, getElement, "N"), check.names = FALSE)
rm(Y)
```

#### Filtering

1. read count-based filter with threshold $t_\mathrm{rc}=15$
1. individual-based filter with threshold $t_\mathrm{ind}=25$

The code was copied from an earlier post and is hidden here.
```{r cache=TRUE, echo=FALSE}
min.obs <- 25 # reset t_ind
# implementation detail!: filter out genes with fewer observations than 'min.obs'
g.passed <- names(S)[sapply(S, function(y) sum(! is.na(y)) >= min.obs)]
min.reads <- 15 # set t_rc
S <- filter.min.read(min.reads, X = S[g.passed], N = N[g.passed], min.obs = min.obs)
N <- filter.min.read(min.reads, X = N[g.passed], N = N[g.passed], min.obs = min.obs)
# ECDFs for all filter levels and all genes g; individual ECDF components F_g are named according to gene g
# the expression also sorts genes g according to F_g(0.9) where F_g is the ECDF for gene g 
ED <- list(fun = sorted.ecdfs(S))
# evaluate ECDF at
ED$ss <- seq(0.5, 1, length.out = 101)
ED$val <- data.frame(lapply(ED$fun, function(f) f(ED$ss)), check.names = FALSE)
# fractions of interest
frac <- do.fractions(ED$fun, S, N, frac = 10:6 / 10,
                     ucl.fun = CI.p, max.ucl = 0.7, max.s = 0.6)
# sort data according to gene ranking
S <- S[names(ED$fun)]
N <- N[names(ED$fun)]
```

### Fitting models to selected genes

Import read count data but do **not** filter, to be consistent with the most recent regression analysis:
```{r cache=FALSE, warning=FALSE}
genes2fit <- unlist(read.csv("../../data/genes.regression.new", as.is = TRUE))
names(genes2fit) <- genes2fit
E <- get.predictors() # default arguments
Y <- get.readcounts(gene.ids = genes2fit, count.thrs = 0)
```

Fitting all models to all retained gene-wise and aggregated read count data sets
```{r cache=FALSE, warning=FALSE}
# exclude unweighed aggregates UA.8 and UA from fitting
ids2fit <- grep("^UA(.8)?$", names(Y), value = TRUE, invert = TRUE)
M <- do.all.fits(Y[ids2fit], preds = e.vars)
f.ids <- as.data.frame(lapply(M, function(m) ! sapply(m, is.null)))
f.ids["TMEM261P1", c("logi.S", "logi2.S")] <- FALSE
f.ids[c("WA.8", "WA"), ] <- FALSE # gene aggregates are not needed here
```

## Analysis

### Delineation of imprinted gene clusters

Let $K$ denote the set of genes $g$ such that $g\in K$ means the gene is either "known" (to be imprinted) or near some "known", i.e. "nearby candidate". Let $C$ be the set of "candidate" genes (further away from "known"s). Let $g_{i-1}$ and $g_i$ be neighboring genes on the same chromosome at the $i-1$-th and $i$-th site.  Delineation of clusters was done using the following simple rule: if $g_{i-1}\in C$ and $g_{i}\in K$ then a cluster starts at the $i$-th gene.  The cluster ends at the first $j\gt i$ such that $g_{j}\in K$ but $g_{i}\in C$.  The rule is implemented in the `make.impr.segs` function in `2016-08-08-imprinted-gene-clusters.R`.

```{r cache=TRUE}
gene.summary$imprinting.status <- factor(gene.summary$imprinted, ordered = TRUE)
levels(gene.summary$imprinting.status) <- rev(c("known imprinted", "nearby candidate", "distant candidate"))
gene.summary$Symbol <- factor(gene.summary$Symbol, levels = gene.summary$Symbol, ordered = TRUE)
gene.summary$chr <- factor(paste("chr", gene.summary$chr), levels = paste("chr", seq_along(levels(factor(gene.summary$chr)))), ordered = TRUE)
# imprinting segments in component 'seg': clusters and segments inbetween
gs.seg <- make.impr.segs(gene.summary, remove.str = "distant candidate")
gs.seg$cluster <-
    factor(x <- with(gs.seg,
                     ifelse(seg > 0, paste0("clus ", seg, " (", chr, ")"), paste0("inter clus ", abs(seg)))),
           levels = unique(x), ordered = TRUE)
rm(x)
# remove filtered genes
gs <- gs.seg[names(frac), ]
gs$score <- unlist(frac["1", ])
```

Eeach cluster has several genes including the "nearby candidate" category; note the median as well.

```{r cluster-sizes, echo=TRUE}
cluster.freq <- table(gs.seg$cluster)[seq(2, length(levels(gs.seg$cluster)), by = 2)]
median(cluster.freq)
barchart(cluster.freq, xlab = "# genes in cluster (including nearby candidates)")
```

*Before* filtering these clusters contain the following number of "known" and "nearby candidate" genes:
```{r}
table(gene.summary$imprinting.status)
```

*After* filtering:
```{r}
table(gs$imprinting.status)
```

### Genomic location

The plot below shows the genomic location of all `r nrow(gene.summary)` genes in the filtered data set and indicates their imprinting status with different colors.  Also shown is the gene score according to which genes have been ranked and called as monoallelically expressing or not.

```{r score-genomic-location, dev=c("png", "pdf"), fig.asp=1}
t.par <- list(superpose.symbol = list(pch = c(21, 21, 21), alpha = c(0.3, 1, 1), fill = c("pink", "green", "lightblue"), col = c("red", "darkgreen", "blue")))
trellis.par.set(t.par)
xyplot(score ~ start | chr, data = gs, groups = imprinting.status, auto.key = list(columns = 3), layout = c(4, 6),
       xlab = "genomic location", ylab = "gene score")
```

Extract $\hat{\beta}_\mathrm{age}$ and confidence intervals for $\beta_\mathrm{age}$ 

```{r warning=FALSE, message=FALSE}
beta.age <- list()
beta.age$logi.S <- do.beta.age(M$logi.S[f.ids$logi.S])
beta.age$logi2.S <- do.beta.age(M$logi2.S[f.ids$logi2.S])
beta.age$wnlm.R <- do.beta.age(M$wnlm.R[f.ids$wnlm.R])
beta.age$wnlm.Q <- do.beta.age(M$wnlm.Q[f.ids$wnlm.Q])
```

The next plot presents the maximum likelihood estimate $\hat{\beta}_\mathrm{age}$ of the regression coefficient mediating age's effect in the logistic model `logi.S`.  Only those genes are shown that were fitted, of course.  Confidence intervals are not shown for clarity.

```{r beta-genomic-location, dev=c("png", "pdf"), fig.asp=1}
gs$beta.hat <- NA
gs[rownames(beta.age$logi.S), "beta.hat"] <- beta.age$logi.S$beta.hat
xyplot(beta.hat ~ start | chr, data = gs, groups = imprinting.status, auto.key = list(columns = 3), layout = c(4, 6), panel = function(...) { panel.abline(h = 0, col = trellis.par.get("reference.line")$col); panel.xyplot(...) }, xlab = "genomic location", ylab = expression(beta[age]))
```

### Clusters and the age effect

#### 99 % confidence

```{r echo=FALSE}
my.segplot <- function(data, ...) {
segplot(Gene ~ Lower.CL + Upper.CL | cluster, data = data,
               # groups = imprinting.status, # groups does not seem to work with segplot
               scales = list(y = list(relation = "free", rot = 0)),
               panel = function(...) {
                   panel.grid(v = -1, h = 0)
                   panel.abline(v = 0, col = "red")
                   panel.segplot(...)
               },
               draw.bands = FALSE, centers = beta.hat,
               #as.table = FALSE,
               layout = c(1, length(levels(data$cluster))),
               par.settings = list(layout.heights = list(panel = (x <- table(data$cluster))[x > 0]),
                                   add.text = list(cex = 0.8)),
               xlab = expression(beta[age]),
               ...)
}
```

After some uninteresting data manipulation (code hidden) **the main result** can be presented at 99 % confidence:

```{r segplot-4-models, fig.asp=2, echo=FALSE, out.height=1400}
sp <- list()
sp$logi.S <- my.segplot(beta.age$logi.S, main = "logi.S", xlim = c(-0.04, 0.04))
sp$logi2.S <- my.segplot(beta.age$logi2.S, main = "logi2.S", xlim = c(-0.04, 0.04))
sp$wnlm.Q <- my.segplot(beta.age$wnlm.Q, main = "wnlm.Q", xlim = c(-0.7, 0.7))
sp$wnlm.R <- my.segplot(beta.age$wnlm.R, main = "wnlm.R", xlim = c(-0.7, 0.7))
plot(sp$logi.S, split = c(1, 1, 2, 2), more = TRUE)
plot(sp$logi2.S, split = c(2, 1, 2, 2), more = TRUE)
plot(sp$wnlm.R, split = c(1, 2, 2, 2), more = TRUE)
plot(sp$wnlm.Q, split = c(2, 2, 2, 2), more = FALSE)
```

#### 95 % confidence

```{r warning=FALSE, message=FALSE}
beta.age.95 <- list()
beta.age.95$logi.S <- do.beta.age(M$logi.S[f.ids$logi.S], conf.lev = 0.95)
beta.age.95$logi2.S <- do.beta.age(M$logi2.S[f.ids$logi2.S], conf.lev = 0.95)
beta.age.95$wnlm.R <- do.beta.age(M$wnlm.R[f.ids$wnlm.R], conf.lev = 0.95)
beta.age.95$wnlm.Q <- do.beta.age(M$wnlm.Q[f.ids$wnlm.Q], conf.lev = 0.95)
```

As expected, the age effect is significant for more genes at lower confidence

```{r segplot-4-models-95, fig.asp=2, echo=FALSE, out.height=1400}
sp.95 <- list()
sp.95$logi.S <- my.segplot(beta.age.95$logi.S, main = "logi.S", xlim = c(-0.04, 0.04))
sp.95$logi2.S <- my.segplot(beta.age.95$logi2.S, main = "logi2.S", xlim = c(-0.04, 0.04))
sp.95$wnlm.Q <- my.segplot(beta.age.95$wnlm.Q, main = "wnlm.Q", xlim = c(-0.02, 0.02))
sp.95$wnlm.R <- my.segplot(beta.age.95$wnlm.R, main = "wnlm.R", xlim = c(-0.7, 0.7))
plot(sp.95$logi.S, split = c(1, 1, 2, 2), more = TRUE)
plot(sp.95$logi2.S, split = c(2, 1, 2, 2), more = TRUE)
plot(sp.95$wnlm.R, split = c(1, 2, 2, 2), more = TRUE)
plot(sp.95$wnlm.Q, split = c(2, 2, 2, 2), more = FALSE)
```


#### Figure for the manuscript

at 99 % confidence:

```{r segplot, fig.width=3.5, fig.asp=2.25, out.height=1000, echo=FALSE}
update(sp$logi.S, main = "")
```

## Conclusion

The last, main, result suggests that

1. age operates on genes within the same cluster independently rather than in concert
1. however, many results are "weak":
   * relatively low power indicated by several extremely wide confidence intervals
   * the fit of the logistic model is not very satisfactory (discussed earlier, see conditional analysis)
