<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
  <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
</head>
<body>
<h1 id="more-discussion-on-manuscript">More discussion on manuscript</h1>
<h2 id="overall">Overall</h2>
<ul>
<li>What are the most interesting findings?</li>
</ul>
<blockquote>
<p>Andy: (1) There are seem to be much fewer imprinted genes as some prev. studies suggested (Catherine Dulac et al). (2) Age dependence of imprinting. (3) First(?) study using human samples.</p>
</blockquote>
<ul>
<li>What needs further analysis?
<ul>
<li>regression on age of death: model selection?</li>
<li>Error rates for calling monoallelic expression necessary?</li>
</ul></li>
</ul>
<blockquote>
<p>We started discussing the necessity of error control for supporting the first conclusion above. I will write a more formal account on error control soon.</p>
</blockquote>
<h2 id="regression">Regression</h2>
<p><span class="math inline">\(\mathrm{E}[ \mathrm{LOI\_R} ] = x_1 \beta_1 + ...\)</span>, where <span class="math inline">\(x_1\)</span> is the age of death and the null hypothesis is <span class="math inline">\(\beta_1=0\)</span>, i.e. age has no impact on imprinting. <span class="math inline">\(\mathrm{LOI\_R}\)</span> seems to be a variable that in some way aggregates <span class="math inline">\(\{S_g\}_g\)</span> over 8 or 13 genes <span class="math inline">\(g\)</span>. But what is the definition of <span class="math inline">\(\mathrm{LOI\_R}\)</span>? What kind of aggregation is it (summation, pooling,...)?</p>
<blockquote>
<p>We looked at Ifat's R script for regression analysis and the definition of LOI_R. When I get access to her files on the other server I'll look at them.</p>
</blockquote>
<h2 id="error-rates">Error rates</h2>
<p>The manuscript provides no error rates for the classification of genes as mono or biallelically expressed.</p>
<ol style="list-style-type: decimal">
<li>frequentist approach
<ul>
<li><span class="math inline">\(p\)</span>-values based on the null distribution of <span class="math inline">\(S_g^{(i)}\)</span></li>
<li>FDR control based on estimate of fraction of monoallelically expressed genes</li>
</ul></li>
<li>Bayesian approach
<ul>
<li>probabilities of mono/biallelic expression: <span class="math inline">\(\pi(m) + \pi(b) = 1\)</span></li>
<li>prior prob. based on
<ol style="list-style-type: lower-alpha">
<li>estimate of fraction of monoallelically expressed genes</li>
<li>distance from known imprinted genes <span class="math inline">\(\pi(m | \mathrm{dist})\)</span> (further extension: HMM)</li>
</ol></li>
<li>posterior <span class="math inline">\(\pi(m | x, y)\)</span> given
<ol style="list-style-type: lower-alpha">
<li>expression data <span class="math inline">\(x\)</span> (RNA-seq)</li>
<li>genotype data <span class="math inline">\(y\)</span> (SNP-array)</li>
<li>likelihood <span class="math inline">\(f(x,y|m)\)</span> for <span class="math inline">\(m\)</span> based on <span class="math inline">\(x,y\)</span></li>
<li>prior <span class="math inline">\(\pi(m)\)</span></li>
</ol></li>
</ul></li>
</ol>
<h3 id="notes">Notes</h3>
<p>In the frequentist approach we only need the likelihood function <span class="math inline">\(f(x,y|b)\)</span>, whereas in the Bayesian one we also need <span class="math inline">\(f(x,y|m)\)</span> (and the prior <span class="math inline">\(\pi\)</span>, of course).</p>
<blockquote>
<p>Andy: permutation-derived null distribution of <span class="math inline">\(S_g^{(i)}\)</span> seems preferable instead of binomial assumption</p>
</blockquote>
<p>The form of likelihood <span class="math inline">\(f\)</span> depends on the dependency structure of the following variables:</p>
<h3 id="error-of-genotype-calling">Error of genotype calling</h3>
<p>A different kind of error rates <strong>is</strong> provided: error for calling genotypes (Figures <a href="http://katahdin.mssm.edu/ifat/web/cm/figures/error1.html">error rate 1</a> and <a href="http://katahdin.mssm.edu/ifat/web/cm/figures/error2.html">error rate 2</a>.</p>
<blockquote>
<p>Andy: discordant call is when RNA-seq suggests monoallelic expression and the Chip-array suggests heterozygosity</p>
</blockquote>
<p>Figure <a href="http://katahdin.mssm.edu/ifat/web/cm/figures/error1.html">error rate 1</a>.</p>
<ul>
<li>Does <code>AB</code> mean heterozygosity for a given SNP?</li>
<li><code>all calls</code>: within an individual? For all individuals?</li>
<li>How was <code>probability for AB</code> calculated? What is <code>chip</code>?</li>
<li>What are <code>err_AB_100</code> and <code>err_AB_7</code>?</li>
</ul>
<blockquote>
<p>Andy: the acceptable minimum number of fragments covering a given SNP.</p>
</blockquote>
<p>Figure <a href="http://katahdin.mssm.edu/ifat/web/cm/figures/error2.html">error rate 2</a>.</p>
<ul>
<li>What are <code>data points</code>?</li>
<li>What are numbers (1 to 100) next to plotted symbols?</li>
</ul>
<h2 id="association-of-hla-genes-to-schizophrenia">association of HLA genes to schizophrenia</h2>
<p>Nonsignificant tendency for HLA-DQB1 was found. Worth following up?</p>
<blockquote>
<p>Andy: not really worth it.</p>
</blockquote>
<p>Imputation of HLA types uses two sources of info</p>
<ol style="list-style-type: decimal">
<li>SNPs (HIBAG)</li>
<li>RNA-seq (PHLAT)</li>
</ol>
</body>
</html>
