---
layout: default
title: Genome-wide Extent of Monoallelic Expression
tags: [genome, monoallelic-expression, manuscript]
featimg: "complex-plot-b-1.png"
---

Various plots present the distribution of read count ratio across individuals and genes jointly.  Genes are also ranked using a summary statistic derived from the empirical distribution of read count ration across individuals for each given gene.

### Missing data due to bug

For each gene $$g$$ of 16026 genes the higher and lower read counts across $$I_g\le 579$$ individuals $$\{H_{ig}, L_{ig} \,\mid \, i=1,...,I_g\}$$ are stored in a `.csv` file.  However, nearly 3 % of those files are empty due to a bug that resulted in multiple headers in [Ifat's html tables][ifat] (see for instance gene AK3) and to the inability of my converter script to deal with such tables.  To see this:

```{r engine="bash"}
DATADIR="../../data/readcount/"
SEDCMD='s/^.*attila attila\s\+\([[:digit:]]\+\).*\/\([^/]\+\)\.csv$/\1,\2/'
OUTFILE="${DATADIR}/fsize-genes.csv"
[ -f $OUTFILE ] || {
    # create header
    echo "file.size,gene.symbol" > $OUTFILE
    # get file size (in bytes) for each gene under 'genes' directory
    find "${DATADIR}/genes" -name '*.csv' | xargs ls -l | sed "$SEDCMD" >> $OUTFILE
}
cat <<EOF
Number of .csv files

all: $(grep --count '^[[:digit:]]\+,' $OUTFILE )
empty: $(grep --count '^0,' $OUTFILE )
EOF
```

```{r echo=FALSE}
library(lattice)
library(latticeExtra)
opts_chunk$set(dpi = 144)
opts_chunk$set(out.height = "700px")
opts_chunk$set(dev = c("png", "pdf"))
lattice.options(default.args = list(as.table = TRUE))
lattice.options(default.theme = "standard.theme")
lattice.options(skip.boundary.labels = 0)
```

### Import read count data

Load data importer functions:
```{r}
source("../../src/import-data.R")
source("../../src/utils.R")
source("2016-07-19-genome-wide-S.R")
```

The following expressions import $$S_{ig}$$ for all `r 16026 - 442` genes for which the csv file is nonempty.

```{r cache=TRUE}
gene.summary <-
    read.csv("../../data/readcount/summary-all-genes.csv")
rownames(gene.summary) <- gene.summary$Symbol
gene.summary$file.size <-
    read.csv("../../data/readcount/fsize-genes.csv", row.names = 2)[rownames(gene.summary), , drop = TRUE]
gene.ids <- with(gene.summary, as.character(Symbol)[ file.size > 0 ])
```

```{r genomwide-data-import, cache=TRUE, warning=FALSE}
Y <- get.readcounts(gene.ids, g.subsets = list(), sel.var = c("H", "N"), rm.conflict = TRUE)
S <- data.frame(lapply(gene.ids, function(g) Y[[g]]$H / Y[[g]]$N), check.names = FALSE)
names(S) <- gene.ids
N <- data.frame(lapply(Y, getElement, "N"), check.names = FALSE)
rm(Y)
```

Perform filtering

### Filtering genes based on number of observations

For more than half of even the genes $$g$$ with nonempty files the number $$I_g$$ of observations (the number of individuals/RNA samples with read count data on $$g$$) is zero.  In what follows, not only these genes are filtered out but also those with less than 10 observations, indicated by the vertical dashed line on the empirical ECDF plot below.

```{r}
min.n.obs <- 25
```
```{r ecdf-n-obs, echo=FALSE}
ecdfplot(sapply(S, function(y) sum(! is.na(y))), type = c("l", "g"),
         panel = function(...) {
             panel.ecdfplot(...)
             panel.abline(v = min.n.obs, lty = 2)
         }, xlab = "number of observations on gene",
         main = paste0("Read count availability (", as.character(ncol(S)), " genes)"))
```

### Data preparation

Update: **filtering** based on a subsequent post
```{r cache=TRUE}
min.obs <- 25 # reset t_ind
# implementation detail!: filter out genes with fewer observations than 'min.obs'
g.passed <- names(S)[sapply(S, function(y) sum(! is.na(y)) >= min.obs)]
min.reads <- 15 # set t_rc
S <- filter.min.read(min.reads, X = S[g.passed], N = N[g.passed], min.obs = min.obs)
N <- filter.min.read(min.reads, X = N[g.passed], N = N[g.passed], min.obs = min.obs)
# ECDFs for all filter levels and all genes g; individual ECDF components F_g are named according to gene g
# the expression also sorts genes g according to F_g(0.9) where F_g is the ECDF for gene g 
ED <- list(fun = sorted.ecdfs(S))
# evaluate ECDF at
ED$ss <- seq(0.5, 1, length.out = 101)
ED$val <- data.frame(lapply(ED$fun, function(f) f(ED$ss)), check.names = FALSE)
# fractions of interest
frac <- do.fractions(ED$fun, S, N, frac = 10:6 / 10,
                     ucl.fun = CI.p, max.ucl = 0.7, max.s = 0.6)
# sort data according to gene ranking
S <- S[names(ED$fun)]
N <- N[names(ED$fun)]
```

Computationally demanding calculations to prepare data for presentation:

```{r genomwide-data-shaping-1, cache=TRUE, warning=FALSE}
ED.long <- reshape(ED$val, v.names = "ECDF", varying = names(ED$val),
                   timevar = "gene", times = factor(names(ED$val)),
                   idvar = "s", ids = ED$ss, direction = "long")
ED.long$gene <- factor(ED.long$gene, levels = names(ED$val), ordered = TRUE)
```

### Figure for manuscript

This figure is intended to:

* present a few representative genes characterized by the "read count ratio" statistic $$S$$ (two upper graphs)
  * e.g. PEG10 and ZNF331 represent monoallelic expression and AFAP1 biallelic expression
  1. density est.: kernel density estimates, whose interpretation is identical to that of histograms
  1. ECDF: empirical cumulative distribution function, sometimes called cumulative fraction
* present a compact genome-wide overview of parental imbalance (lower left graph)
  * imbalance is expressed in terms of the ECDF of $$S$$ using a color scheme, introduced in the upper 2nd and 3rd graphs
  * the thousands of genes are ranked from the most imbalanced (monoallelic, top) to the most balanced (biallelic, bottom)
  * the ranking is based on the ECDF evaluated at $$s = 0.9$$, as shown in the upper 2nd graph and the bottom right plots
* support the conclusion that $$\approx 1 \%$$ of all genes are appreciably imbalanced (monoallelically expressed)

```{r cache=TRUE, echo=FALSE, warning=FALSE}
Y <- get.readcounts(gene.ids, g.subsets = list(), sel.var = c("N"))
```

There are `r length(S)` genes on this figure but before filtering there were `r length(Y)` genes.

```{r complex-plot, echo=FALSE, fig.asp=1.00}
sel.g <- c("PEG10", "ZNF331", "AFAP1")
S.long <- reshape.Y(Y = S, sel.g = sel.g)
ramp.cols <- c("red", "blue")
plots <- list()
#superpose.line <- standard.theme(color = FALSE)$superpose.line
#superpose.line$lty <- c(2, 1, 3:7)
plots$density <- my.densityplot(S.long, par.settings = list(), xlab = NULL, ylab = "density")
plots$density <- update(plots$density, scales = list(y = list(at = pretty(density(S[[sel.g[1]]], na.rm = TRUE)$y, n = 3))))
plots$ecdf <- my.ecdfplot2(S.long, invert.col.grad = TRUE, ramp.cols = ramp.cols, lwd = 2,
                           par.settings = list(plot.symbol = list(pch = 21, col = "black", fill = "lightgreen", cex = 2)))
plots$level <- my.levelplot(ED.long, pct.top.g = 2, n.all.g = length(levels(ED.long$gene)), top.on.top = TRUE, xlab = "read count ratio, S", ylab = "gene rank", col.regions = colorRampPalette(ramp.cols, 100))
plots$level.sel.g <- my.levelplot.sel.g(ED.long, sel.g = sel.g, letter.labels = TRUE, col.regions = colorRampPalette(ramp.cols, 100))
plots$rank <- rankplot.2(frac, pct.top.g = 2, sel.g = sel.g, top.on.top = TRUE,
                          xlab = "gene score", ylab = NULL,
                         par.settings = list(superpose.symbol =
                                             list(pch = 21, col = c("black", "black"), fill = c("lightgray", "lightgreen"), cex = c(1.0, 2))))
# print to complex layout
trellis.par.set(col.whitebg())
plot.all.c(plots)
```

```{r complex-plot-density, echo=FALSE, fig.asp=1.25}
plot.all(plots)
```

### Figures for live presentation

```{r complex-plot-b, echo=FALSE, fig.asp=1.0, fig.width=5}
plot.all.c(plots)
plot.all.b(plots)
```

```{r complex-plot-c, echo=FALSE, fig.asp=1.5, out.height=1000, fig.width=5}
plot.all.c(plots)
plot.all.b(plots)
```

## Revision for Nature Communications

```{r}
source("../../src/fit-glms.R")
set.seed(19760415) # my birthdate
sel.g.rnd <- sample(names(S), 42)
Y.long <- reshape.Y2long(sel.g.rnd)
```

```{r S-Dx-strip, warning=FALSE, echo=FALSE, fig.asp = 1.2}
trellis.par.set(standard.theme())
dp <-
    stripplot(S ~ Dx | Gene, groups = Dx, data = Y.long, subset = Y.long$Dx %in% c("Control", "SCZ", "AFF"),
              plot.points = FALSE, panel = function(x, y,...) {
                  if(FALSE) # violin plot doesn't work for several data pockets
                      panel.violin(x, y,..., col = "gray", border = 0)
                  panel.bwplot(x, y,..., pch = "|")
                  panel.stripplot(x, y,..., jitter.data = TRUE, factor = 4, pch = 21, cex = 0.3)
              })
my.update <- function(dp, ...) {
    update(dp, scales = list(x = list(draw = FALSE, relation = "free")),
           par.settings = list(add.text = list(cex = 0.8)),
           xlab = "read count ratio, S",
           ...)
}
update(my.update(dp), scales = list(x = list(draw = TRUE, relation = "same", rot = 90)),
       xlab = "Dx", ylab = "read count ratio, S", auto.key = list(columns = 3),
       par.settings = list(box.rectangle = list(col = "black", fill = "gray"), box.umbrella = list(col = "black"), plot.symbol = list(cex = 0)), layout = c(6, 7))
```


[ifat]: http://katahdin.mssm.edu/ifat/web/cm/home
[Fig 1]: https://docs.google.com/presentation/d/1YvpA1AJ-zzir1Iw0F25tO9x8gkSAzqaO4fjB7K3zBhE/edit#slide=id.p4
