---
layout: default
tags: [ andy, chaggai, manuscript ]
---

Following a discussion with Andy and Chaggai, and having received feedback from Doug, the following changes have been made to the manuscript:

1. All figures and tables have been moved below the References section to facilitate submission to journals
1. Figure S1 (Clustering of top-scoring genes) has now a *fourth* gene category called *below top 50*.  In conjunction with this, the term *candidate* has been redefined to mean genes in the top 50 that were not known previously to be imprinted.
1. The original Figure 4 (Variation of read count ratio) has been duplicated to Fig 4 and a new supplementary Fig.  The new Fig 4 has been changed so that gender information is *not* shown, so that Fig 4 has become grayscale.  The main text and figure legends have been updated accordingly.
1. Figure 7: I clarified the scaling in the figure legend by inserting the sentence: "*Note that the coefficients for different predictors
are on different scale in general therefore it is not meaningful to compare them across
graphs.*"
1. The following sentence has been added: "Note that our later analyses (Section 3.2 and below) used information not only in \(S_{ig}\) but also in the total read count as well as in data beyond RNA-seq."
1. The justification of selecting UBE3A was further clarified: "In subsequent analysis (below) we also consider UBE3A as demonstrating allelic bias consistent with imprinting in the context of human adult DLFPC as evidenced by Fig.~\ref{fig:known-genes} even though its rank falls outside of the top 50."
1. Fig 1 has been updated
    * silhouettes of individuals have been removed
    * RNA molecules and RNA-seq reads have been annotated
    * legend has been simplified and improved

Suggested changes that have not been undertaken:

* Supplementary Figure 13 (Significance of association...), formerly Fig. S12, remained unchanged despite the suggestion to remove those symbols that represent genes for which the fit was excluded (or failed) under the logi.S model.  Justification: the shape of the symbols already informs on whether logi.S was also fitted for a given gene.  Moreover, removal of the said symbols would require substantial programming.  I doubt this minor point would be worth the effort.
