\documentclass[usenames,dvipsnames]{beamer} %\documentclass[handout]{beamer}


%\includeonlyframes{bsm,title,toc,imprint-mouse-devel,igf2-imprint-evol,sister-disorders,previous-age-studies,our-study,cmc,toc-current,filtering-calling,fitting-models,ll-surface,all-betas,orthogonality,identifiability,anova,betas-cluster,signif-genes,cf-overall-expression,summary,improving-model,p-val,imprinted-brain,chess-lab}

% language settings
%\usepackage{fontspec, polyglossia}
%\setdefaultlanguage{magyar}

% common packages
\usepackage{amsmath, multimedia, hyperref, color, multirow}
%\usepackage{graphicx}

% TikZ
\usepackage{tikz}
%\usetikzlibrary{arrows.meta, decorations.pathmorphing, decorations.pathreplacing, shapes.geometric,mindmap}
%\usetikzlibrary{shapes.geometric,fadings,bayesnet}

% beamer styles
\mode<presentation>{
%\usetheme{Pittsburgh}
\usetheme{Singapore}
%\usecolortheme{dove}
\usecolortheme{seahorse}
%\usefonttheme{structureitalicserif}
\setbeamercovered{transparent}
}
\setbeamertemplate{blocks}[rounded][shadow=true]
\AtBeginSubsection[]{
  \begin{frame}<beamer>{Contents}
    %\tableofcontents[currentsection]
    \tableofcontents[currentsubsection]
  \end{frame}
}
%\useoutertheme[]{tree}

\title{Imprinting, schizophrenia, and aging}
\subtitle{investigated with mixed models and CommonMind data}
\author{Attila Guly\'{a}s-Kov\'{a}cs, Chess lab}
%\date{Chess Lab}
\date{\includegraphics[height=0.5\textheight]{figures/explain-rll-wireframe-1.png}}

\newcommand{\platefigscale}[0]{0.7}
\newcommand{\ownfigscale}[0]{0.4}

\begin{document}

\begin{frame}[plain, label=title]
\maketitle
\end{frame}

\begin{frame}{Contents}
\tableofcontents
\end{frame}

\section{Introduction}

\begin{frame}{Schizophrenia: complex genetic architecture}
\includegraphics[width=1.0\textwidth]{figures/from-others/sullivan-natrevgenet-2012-fig1b.jpg}
\end{frame}

\begin{frame}{Imprinting and allelic bias}
\begin{columns}[t]
\begin{column}{0.5\textwidth}
\begin{enumerate}
\item roles
\begin{itemize}
\item psychiatric?
\item mother--child interaction
\item neurodevelopmental 
\end{itemize}
\item variation: age, tissue
\end{enumerate}
\end{column}

\begin{column}{0.5\textwidth}

\includegraphics[width=1.0\columnwidth]{figures/from-others/renfree-2012-fig2.jpg}

{\tiny Renfree et al 2012 Philos Trans R Soc Lond B}
\end{column}
\end{columns}
\end{frame}

\begin{frame}{Our \only<1>{\emph{previous}} \only<2>{\emph{revised}} work}
\begin{itemize}
%\item genomic imprinting and schizophrenia (SCZ)
\item<1> CommonMind Consortium data
\item<1> 30 imprinted genes in DLPFC, some novel
\item<1> found allelic bias to depend on SCZ, age
\begin{itemize}
\item genes varied
\item weak statistical support 
\end{itemize}
\item<2> mixed models: more power
\item<2> found allelic bias to depend on age but \alert{not} SCZ
\begin{itemize}
\item genes varied
\end{itemize}
\end{itemize}
\end{frame}

\section{Our previous work}

\begin{frame}{Read count ratio measures allelic expression bias}
\includegraphics[height=0.75\textheight]{figures/by-me/commonmind-rna-seq/commonmind-rna-seq.pdf}
\end{frame}

\begin{frame}{The data}
\footnotesize
\begin{tabular}{|c|ccc|cccc|}
 & \multicolumn{3}{|c|}{read count ratios} & \multicolumn{4}{|c|}{explanatory variables} \\
\(i\) & \(Y_1\) & \(\hdots\) & \(Y_{30}\) & \(X_1\) & \(X_2\) & \(\hdots\) & \(X_{12}\) \\
\textbf{Individual} & \textbf{MAGEL2} & \(\hdots\) & \textbf{UBE3A} & \textbf{Dx} & \textbf{Age} & \(\hdots\) & \textbf{RIN} \\
\hline
1 & NA & \(\hdots\) & NA & AFF & 42 & \(\hdots\) & 6.90 \\
2 & 1.00 & \(\hdots\) & NA & AFF & 58 & \(\hdots\) & 7.00 \\
3 & NA & \(\hdots\) & 0.89 & AFF & 28 & \(\hdots\) & 6.90 \\
\(\vdots\) & \(\vdots\) & \(\vdots\) & \(\vdots\) & \(\vdots\) & \(\vdots\) & \(\vdots\) & \(\vdots\) \\
%574 & NA & \(\vdots\) & 0.90 & SCZ & 59 & \(\vdots\) & 8.60 \\
%575 & NA & \(\vdots\) & NA & Control & 38 & \(\vdots\) & 8.30 \\
576 & 0.97 & \(\hdots\) & NA & Control & 42 & \(\hdots\) & 8.50 \\
577 & 1.00 & \(\hdots\) & NA & SCZ & 27 & \(\hdots\) & 7.50 \\
578 & 1.00 & \(\hdots\) & NA & Control & 57 & \(\hdots\) & 8.60 \\
579 & 1.00 & \(\hdots\) & 1.00 & Control & 28 & \(\hdots\) & 7.70 \\
\end{tabular}
\end{frame}

\begin{frame}%[plain]
\begin{center}
\includegraphics[height=0.95\textheight]{figures/2016-11-01-plotting-distribution-of-s/S-Dx-strip-1.pdf}
\end{center}
\end{frame}

\begin{frame}%[plain]
\begin{center}
\includegraphics[height=0.95\textheight]{figures/2016-06-26-trellis-display-of-data/evar-scatterplot-matrix-simple-1.pdf}
\end{center}
\end{frame}

\begin{frame}[label=S-Age]
\begin{center}
\includegraphics[height=0.95\textheight]{figures/2016-11-01-plotting-distribution-of-s/S-Dx-age-1.pdf}
\end{center}
\end{frame}

\begin{frame}
\begin{center}
\includegraphics[height=0.95\textheight]{figures/2018-02-10-fixed-mixed-p-values/beta-99-CI-1.pdf}
\end{center}
\end{frame}

\begin{frame}<1,2>[label=pval]
\begin{center}
\includegraphics<1>[height=0.95\textheight]{figures/2018-02-10-fixed-mixed-p-values/p-val-dotplot-wnlmQ-1.pdf}
\includegraphics<2>[height=0.95\textheight]{figures/2018-02-10-fixed-mixed-p-values/p-val-dotplot-wnlmQ-unlmQ-1.pdf}
\includegraphics<3>[height=0.95\textheight]{figures/2018-02-10-fixed-mixed-p-values/p-val-dotplot-wnlmQ-mixed-1.pdf}
\end{center}
\end{frame}

\begin{frame}
\begin{center}
\includegraphics[height=0.95\textheight]{figures/2016-09-23-model-checking/res-std-dev-wnlmQ-unlmQ-FAM50B-MEST-1.pdf}
\end{center}
\end{frame}

\section{Mixed models}

\againframe{S-Age}

\begin{frame}%{The power of mixed models}

\begin{columns}[t]
\begin{column}{0.70\textwidth}

\includegraphics<1>[width=\columnwidth]{figures/2017-10-16-fixed-and-mixed-models/data-1.pdf}

\includegraphics<2>[width=\columnwidth]{figures/2017-10-16-fixed-and-mixed-models/data-F0-F1-1.pdf}

\includegraphics<3>[width=\columnwidth]{figures/2017-10-16-fixed-and-mixed-models/p-val-F1F0.pdf}

\includegraphics<4>[width=\columnwidth]{figures/2017-10-16-fixed-and-mixed-models/data-M0-M1-1.pdf}

\includegraphics<5>[width=\columnwidth]{figures/2017-10-16-fixed-and-mixed-models/p-val-F1F0-M1M0.pdf}

\includegraphics<6>[width=\columnwidth]{figures/2017-10-16-fixed-and-mixed-models/data-M0-M1-M2-1.pdf}

\includegraphics<7>[width=\columnwidth]{figures/2017-10-16-fixed-and-mixed-models/p-val-F1F0-M1M0-M2M1-M2M0.pdf}

\includegraphics<8>[width=\columnwidth]{figures/2017-10-16-fixed-and-mixed-models/data-F1-M2-1.pdf}
\end{column}

\begin{column}{0.30\textwidth}

\footnotesize
\begin{tabular}{p{0.30\textwidth} p{0.30\textwidth} p{0.30\textwidth}}
\only<2-8>{fixed} &
\only<4-8>{mixed} &
\only<6-8>{mixed} \\
\only<2-8>{\color{Red}F1} &
\only<4-8>{\color{Green}M1} &
\only<6-8>{\color{Blue}M2} \\
\vspace{0pt}
\includegraphics<2-8>[scale=0.7]{figures/by-me/fixed-mixed/fixed/fixed.pdf} &
\vspace{0pt}
\includegraphics<4-8>[scale=0.7]{figures/by-me/fixed-mixed/mixed/mixed.pdf} &
\vspace{0pt}
\includegraphics<6-8>[scale=0.7]{figures/by-me/fixed-mixed/random/random.pdf} \\
\end{tabular}
\end{column}
\end{columns}
\end{frame}

\begin{frame}
Mixed models and more:\\
https://attilagk.github.io/R-you-experienced/ 
\end{frame}

\section{Our revised work}

\begin{frame}{Fitting mixed models on the CommonMind data}
\begin{columns}[t]
\begin{column}{0.5\textwidth}

\includegraphics[width=\columnwidth]{figures/2017-03-08-model-checking/qqplot-families-M3-1.png}
\end{column}

\begin{column}{0.5\textwidth}

\includegraphics[width=\columnwidth]{figures/2017-03-08-model-checking/scedasticity-families-M3-1.png}
\end{column}
\end{columns}
\end{frame}

\againframe<3>{pval}

\begin{frame}{Summary}
\begin{itemize}
\item caution: model fit
\item<2-> dependence of read count ratio on...
{\footnotesize
\begin{tabular}{r|cc}
explanatory variable & gene nonspecific & gene specific \\
\hline
Schizophrenia (Dx) & - & - \\
Age & - & + \\
Ancestry.{1,3} & - & + \\
Gender & - & + \\
\end{tabular}
}
\item<3> interpretation
\begin{enumerate}
\item no detectable link between imprinting and schizophrenia
\item age, genetics and gender affect imprinting;
genes vary
\end{enumerate}
\end{itemize}
\end{frame}

\begin{frame}{End}
\begin{columns}[t]
\begin{column}{0.5\textwidth}
\begin{itemize}
\item manuscript submitted \\
{\footnotesize
Attila Guly\'{a}s-Kov\'{a}cs, Ifat Keydar,
Eva Xia, Menachem Fromer, Gabriel Hoffman, Douglas Ruderfer,
CommonMind Consortium, Ravi Sachidanandam,
Andrew Chess
}
\item special thanks \\
{\footnotesize Chaggai Rosenbluh}
\item mixed models and more \\
{\footnotesize
https://attilagk.github.io/R-you-experienced/ 
}
\end{itemize}
\end{column}

\begin{column}{0.5\textwidth}

\includegraphics[width=\columnwidth]{figures/2016-08-23-glm-sampling-distributions/GRB10-1.png}
\end{column}
\end{columns}
\end{frame}

\end{document}


\begin{columns}[t]
\begin{column}{0.5\textwidth}

\end{column}

\begin{column}{0.5\textwidth}

\end{column}
\end{columns}



\begin{frame}{Two extreme views}
\begin{columns}[t]
\begin{column}{0.5\textwidth}

differential effects
\begin{itemize}
\item flexible but complex
\end{itemize}

\includegraphics[scale=0.8]{figures/by-me/monoall-dependencies-2/obs-simple-general/obs-simple-general}
\end{column}

\begin{column}{0.5\textwidth}

uniform effects
\begin{itemize}
\item rigid but parsimonious
\end{itemize}

\includegraphics[scale=0.8]{figures/by-me/monoall-dependencies-2/obs-simple-general-gene-aspec/obs-simple-general-gene-aspec}
\end{column}
\end{columns}
\end{frame}

