---
layout: default
title: 'Notes on the "Monoallelic Expression (MAE) Draft"'
tags: [ discussion, monoallelic-expression, andy ]
---

Manuscript title: `Novel monoallelically-expressed genes and relaxation of imprinting with advanced age in the dorsolateral prefrontal cortex.`

| abbreviation | concept |
| ------------- | ------------- |
| (R)MAE | (random) monoallelic expression |
| SCZ | schizophrenia |

## Rationale and expectations

### Functional roles
* in immune and olfactory system clear function: RMAE ensures one cell---one receptor
* hypothetical function of RMAE in brain?
    * MHC association to SCZ => immunity. Mechanism in SCZ?  Autoimmune disease?
    * other roles (beyond immunity)?

> Andy: generate diversity and/or dosage regulation (Dscam paper by [Neves,..., Chess], neuronal identity); mammals neurexin, pcdh protocadherin

* hypothetical function of imprinting?

> Andy: being diploid is generally beneficial but: battle between sexes growth control males boost growth promoting genes (e.g. IGF2); females suppress; Heigh's theory

### Onset and extent of RMAE
* X-inactivation expected to be polyclonal in brain sample
"the unsurprising notion that most samples are polyclonal with respect to X inactivation" (p.5).  Basis of expectation?
* RMAE expected to be monoclonal?  Why?  Developmental onset of RMAE?
* in limphoblastoid cell lines: RMAE is widespread (occurs in 5-10 % of genes, [Gimelbrant et al]) consistent with role in immunity
* known extent of RMAE in other cell lines or tissue types?  Does extent correlate with functional imporatnce in some tissue?

> Andy: 0.7 % indicates that RMAE has less importance in the brain than in the immune systems*

### Molecular mechanisms

* expected lenght of epigenetic marks
    * imprinting
    * RMAE
> Andy: local duplication ORs; H region

## Methodology

* quantity of interest: **allelic imbalance** (avoid skewness: confusing terminology)
* should we care for error rates? Currently heuristics: "To be conservative about calling monoallelic expression" (p.4)
    * are there gold standard data for MAE?
* invest in statistical modeling of allelic imbalance?
    * length of epigenetic marks: correlation between loci (SNPs)
    * dissect sources of variability: across subjects, cell clones/tissues, technical noise (read depth, quality,...)

## Findings

### Extent of MAE

> ms p.3: the discordance rate (monoallelic RNA genotype in the context of a heterozygous DNA genotype) was 0.71%"

how about consistency with 5-10 % ([Gimelbrant et al])?

> Andy: two papers by [Gregg,..., Dulac], which was followed up by [DeVeale et al]

[DeVeale et al]: http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1002600
[Gregg,..., Dulac]: http://classic.sciencemag.org/content/329/5992/682.long
[Neves,..., Chess]: http://dx.doi.org/10.1038/ng1299
[Gimelbrant et al]: http://classic.sciencemag.org/content/318/5853/1136.long
